src/any_agent/evaluation/evaluation_case.py

```diff
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+# This can't go into a TYPE_CHECKING block because it's used at runtime by Pydantic to build the fields
```

Comment:
btw: https://docs.pydantic.dev/latest/concepts/performance/#sequence-vs-list-or-tuple-with-mapping-vs-dict

I think we could type with `list`